src/Symfony/Component/HttpFoundation/RateLimiter/AbstractRequestRateLimiter.php

```diff
@@ -35,7 +35,7 @@ public function consume(Request $request): RateLimit
         foreach ($limiters as $limiter) {
             $rateLimit = $limiter->consume(1);
 
-            if (null === $minimalRateLimit || $rateLimit->getRemainingTokens() < $minimalRateLimit->getRemainingTokens()) {
+            if (null === $minimalRateLimit || $rateLimit->getRemainingTokens() < $minimalRateLimit->getRemainingTokens() || !$rateLimit->isAccepted()) {
```

Comment:
I think we can take this one step further and always return the most "minimal" rate limit. I.e. in case both `$rateLimit` and `$minimalRateLimit` do not accept, return the one with the longest wait time.